apps/homescreen/test/unit/pages_test.js

```diff
@@ -284,10 +284,16 @@ suite('Pages', () => {
           return Promise.resolve(mockEntry);
         });
       storePutStub =
-        sinon.stub(pages.pagesStore.datastore, 'put', (data, id) => {
+        sinon.stub(pages.pagesStore.datastore, 'put', (data, id, revId) => {
           if (callback) {
             callback(data, id);
           }
+          if (revId && storePutStub._raceCondition) {
+            return Promise.reject({
+              name: 'ConstraintError',
+              message: 'RevisionId is not up-to-date'
+            });
```

Comment:
In real life this is a `ConstraintError` object which inherits from `Error`, but just ducktyping it here.